core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java

```diff
@@ -316,6 +327,18 @@ public long getBlockSizeByte(String path) throws IOException {
     return Configuration.getBytes(PropertyKey.USER_BLOCK_SIZE_BYTES_DEFAULT);
   }
 
+  @Override
+  public UfsDirectoryStatus getDirectoryStatus(String path) throws IOException {
+    String keyAsFolder = convertToFolderName(stripPrefixIfPresent(path));
+    ObjectStatus details = getObjectStatus(keyAsFolder);
+    if (details != null) {
+      return new UfsDirectoryStatus(path, getBucketOwner(), getBucketGroup(), getBucketMode());
+    } else {
+      LOG.error("Error fetching file size, assuming file does not exist");
+      throw new FileNotFoundException(path);
```

Comment:
Previously getFileSize would through an exception if path was a directory. It did not attempt both file and directory.